### DISCUSSION TITLE
[ZK Stack] - Change L1 block range

### DISCUSSION THREAD

**Discussion by jishnundth** [zkSync-Community-Hub/zksync-developers#709] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.80.1

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

I'm using chainstack growth plan for L1 RPC, and it has block range capping of 10,000 blocks for eth_getLogs. Is there any way I can change the L1 genesis block while querying the sepolia using chain stack RPC.  I can see that it's always starting querying from 
a block having a range greater than 10,000 to the latest block for sepolia. Even after redeployment it's the same. Is there any config or parameter I can change to adjust this fromBlock for eth_getLogs.

Due to block range capping, the L2 is failing, And 400 is coming from chainstack. 

### Expected Behavior

L2 server should be able to run properly.

### Current Behavior

Unable to process requests.

### Additional Details

`WARN EthWatch::loop_iteration: zksync_eth_watch::client: Provider returned error message: JSON-RPC request get_logs(filter=Filter { from_block: Some(Number(6712839)), to_block: Some(Number(6763059)), block_hash: None, address: Some(ValueOrArray([0x40fdfd51b7635b1c070596432d4dad117953ef0f, 0x06cb09ced0cb57989356fcdfb3e2ebfe9686eb95, 0xa09a0c677f50164fa9a5d672c1603e38f6e62528, 0x3364b7544cc3648a5ab8b2480576f3d309af1554])), topics: Some([Some(ValueOrArray([0x4531cd5795773d7101c17bdeb9f5ab7f47d7056017506f937083be5d6e77a382]))]), limit: None }) failed: ErrorObject { code: InvalidParams, message: "Block range limit exceeded. See more details at https://docs.chainstack.com/docs/limits#evm-range-limits", data: None }`

https://github.com/zkSync-Community-Hub/zksync-developers/discussions/706

This is similar to my current issue. I've redeployed with the mentioned changes, but now the gas price is coming as 0, from RPC. Causing error downstream

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
To address the issue of the block range limit exceeding when using the Chainstack RPC for querying logs on Sepolia, you can adjust the `fromBlock` parameter in your `eth_getLogs` request. This adjustment will help you stay within the 10,000 blocks range limit imposed by Chainstack.

Here are the steps you can follow to modify the `fromBlock` parameter:

1. **Identify the Current Block**: First, determine the current block number on Sepolia. You can do this by querying the latest block directly via your RPC endpoint.

2. **Calculate the `fromBlock` Value**: Set the `fromBlock` to be the current block number minus 9,999. This ensures that your query window does not exceed the 10,000 blocks limit.

3. **Modify the Query in Your Code**: Update the `eth_getLogs` call in your server configuration to use the newly calculated `fromBlock` value. Here is a basic example of how to structure the query:
   ```javascript
   web3.eth.getPastLogs({
       fromBlock: calculatedFromBlock,
       toBlock: 'latest',
       address: [/* array of contract addresses */],
       topics: [/* array of topics you are interested in */]
   });
   ```

4. **Test the Changes**: After making these changes, redeploy your server or restart the service to ensure the new settings are applied. Test to see if the logs are now being fetched within the acceptable range.

5. **Monitor for Errors**: Keep an eye on the logs for any errors similar to the previous ones and adjust the block range if necessary.

By following these steps, you should be able to work within the constraints of the Chainstack RPC service and avoid the block range limit errors, allowing your L2 server to function correctly. If the issue persists, consider reaching out to Chainstack support for more tailored assistance regarding their service limitations.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by shahar4** [comment, marked as answer]
Hey @Jishnu-Nandhiath! I understand that redeploying with the fix mentioned in the other issue has resolved you original issue and now you're facing the exact same issue mentioned [here](https://github.com/zkSync-Community-Hub/zksync-developers/discussions/706#discussioncomment-10772257).

If this is the case, then, unfortunately, the response is the same as in the other thread, we do not support the Chainstack RPC as it doesn't adhere to [Ethereum JSON-RPC Specification](https://ethereum.github.io/execution-apis/api-documentation/).

**Comment by jishnundth** [comment]
Hey @shahar4!  I understood, the issue regarding chain stack. Can you please explain how to change block range while querying the L1 using eth_getLogs

**Reply by shahar4** [reply]
@Jishnu-Nandhiath this isn't manually possible. Please see the context [here](https://github.com/zkSync-Community-Hub/zksync-developers/discussions/706#discussioncomment-10763318) about the internal logic to handle reducing ranges in this component.
